finality-aleph/src/sync/service.rs

```diff
@@ -61,23 +66,27 @@ impl<BI: BlockIdentifier> RequestBlocks<BI> for mpsc::UnboundedSender<BI> {
 }
 
 impl<
+        B: Block,
         J: Justification,
         N: GossipNetwork<VersionedNetworkData<J>>,
         CE: ChainStatusNotifier<J::Header>,
         CS: ChainStatus<J>,
         V: Verifier<J>,
         F: Finalizer<J>,
-    > Service<J, N, CE, CS, V, F>
+        BI: BlockImport<B>,
+    > Service<B, J, N, CE, CS, V, F, BI>
 {
     /// Create a new service using the provided network for communication.
     /// Also returns an interface for submitting additional justifications,
     /// and an interface for requesting blocks.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Nooo~! D: Figure out some wrapping or something?

Reply:
do you want `SyncConfig`, similar to `AlephConfig`?...

Reply:
I'd say let's wait till we have all the necessary building blocks, and only then try to group them, possibly even after refactoring `run_validator_node`

Reply:
I at least wouldn't want this to get forgotten, so a task+TODO pointing to the task here? Although `SyncConfig` would also be good imo.

Reply:
packed